internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,95 @@
+package eventstojobs
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+type EventsToJobService struct {
+	queue                string
+	jobsetid             string
+	jobid                string
+	jobServiceConfig     *configuration.JobServiceConfiguration
+	jobServiceRepository repository.JobServiceRepository
+}
+
+func NewEventsToJobService(
+	queue string,
+	jobsetid string,
+	jobid string,
+	jobServiceConfig *configuration.JobServiceConfiguration,
+	jobServiceRepository repository.JobServiceRepository) *EventsToJobService {
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
+		jobServiceConfig:     jobServiceConfig,
+		jobServiceRepository: jobServiceRepository}
+}
+
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
+
+	return eventToJobService.StreamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+}
+func (eventToJobService *EventsToJobService) StreamCommon(clientConnect *client.ApiConnectionDetails, ctx context.Context) error {
+	var fromMessageId string
+	// This corresponds to the amount of time since the job-set has been updated.
+	// If no new events come from the job-set, then the timer allows automatic unsubscribing.
+	duration := time.Duration(eventToJobService.jobServiceConfig.SubscribeJobSetTime) * time.Second
+	timeOut := time.NewTimer(duration)
+	err := client.WithEventClient(clientConnect, func(c api.EventClient) error {
```

Comment:
I think we need some error handling here to reconnect to the armada event server if necessary.

Reply:
Any code I could use as an example in Armada?